### PR TITLE
Hook goto definition fallback with compiler into LSP.

### PR DESCRIPTION
**Pull request by olafurpg**
A last-minute change in 6c501e51b9d0d0937646e01fdc8a25c26372ed2a made
the goto definition fallback integration with LSP no longer work.  This
PR fixes the issue and adds a test to prevent further regressions.

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala [hunk 1]
```diff
@@ -48,25 +48,27 @@ final class DefinitionProvider(
...
-            None,
-            None
-          )
-        } else {
+    val fromSemanticdb =
+      semanticdbs.textDocument(path).documentIncludingStale match {
+        case Some(doc) =>
+          definitionFromSnapshot(path, params, doc)
+        case _ =>
           warnings.noSemanticdb(path)
```
Comment (gabro) [inline metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala]: does it make sense to warn even in cases in which there's a fallback available?
Mayne the warning could include info in whether Metals has found the fallback or not

Comment (olafurpg) [reply metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala]: I decided to keep the warning and leave it to a separate PR to overall improve the UX (reduce warnings/noise) when semanticdb-scalac is not enabled.

Comment (olafurpg) [reply metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala]: This warning only happens when using "rambo mode" (editing a source file outside a build target).

Comment (olafurpg) [reply metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala]: We don't have the classpath for those source files so I suspect definition/completions won't work very reliably.

Comment (gabro) [reply metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala]: Got it, makes sense.
Incidentally local completions now work in project/*.scala files, which was an unexpected treat 😅

Comment (olafurpg) [reply metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala]: After this PR goto definition will also work there for local variables :)

**Pull request merged**